repository/src/main/java/org/apache/atlas/repository/store/graph/v2/preprocessor/contract/ContractPreProcessor.java

```diff
@@ -0,0 +1,269 @@
+    public void processAttributes(AtlasStruct entityStruct, EntityMutationContext context, EntityMutations.EntityOperation operation) throws AtlasBaseException {
+        AtlasEntity entity = (AtlasEntity) entityStruct;
+        switch (operation) {
+            case CREATE:
+                processCreateContract(entity, context);
+                break;
+            case UPDATE:
+                // Updating an existing version of the contract
+                processUpdateContract(entity, context);
+        }
+
+    }
+
+    private void processUpdateContract(AtlasEntity entity, EntityMutationContext context) throws AtlasBaseException {
+        String contractString = (String) entity.getAttribute(ATTR_CONTRACT);
+        AtlasVertex vertex = context.getVertex(entity.getGuid());
+        AtlasEntity existingContractEntity = entityRetriever.toAtlasEntity(vertex);
+        // TODO: Check for qualifiedName to understand if a particular version is getting updated or duplicate contract in payload
+        if (!isEqualContract(contractString, (String) existingContractEntity.getAttribute(ATTR_CONTRACT))) {
+            // Update the same asset(entity)
+            throw new AtlasBaseException(OPERATION_NOT_SUPPORTED, "Can't update a specific version of contract");
+        }
+        // Add cases for update in status field and certificateStatus
+    }
+    private void processCreateContract(AtlasEntity entity, EntityMutationContext context) throws AtlasBaseException {
+        /*
+          Low-level Design
+               | Authorization
+               | Deserialization of the JSON
+               ---| Validation of spec
+               | Validation of contract
+               | Create Version
+               | Create Draft
+               ---| asset to contract sync
+               | Create Publish
+               ---| two-way sync of attribute
+         */
+
+        String contractQName = (String) entity.getAttribute(QUALIFIED_NAME);
+        validateAttribute(!contractQName.endsWith(String.format("/%s", CONTRACT_QUALIFIED_NAME_SUFFIX)), "Invalid qualifiedName for the contract.");
+
+        String contractString = (String) entity.getAttribute(ATTR_CONTRACT);
+        DataContract contract = DataContract.deserialize(contractString);
+        String datasetQName = contractQName.substring(0, contractQName.lastIndexOf('/'));
+        contractQName = String.format("%s/%s/%s", datasetQName, contract.type.name(), CONTRACT_QUALIFIED_NAME_SUFFIX);
+        AtlasEntityWithExtInfo associatedAsset = getAssociatedAsset(datasetQName, contract.type.name());
+
+        authorizeContractCreateOrUpdate(entity, associatedAsset);
+
+        contractAttributeSync(entity, contract);
+        contractString = DataContract.serialize(contract);
+        entity.setAttribute(ATTR_CONTRACT, contractString);
+
+        ContractVersionUtils versionUtil = new ContractVersionUtils(contractQName, context, entityRetriever, typeRegistry, entityStore, graph, discovery);
```

Comment:
Instead of creating an Object of the class, define getCurrentVersion as static method & pass necessary parameter on the go